packages/dev/core/src/Materials/materialHelper.functions.ts

```diff
@@ -939,8 +943,8 @@ export function PrepareUniformsAndSamplersForLight(
         "depthValues" + lightIndex
     );
 
-    samplersList.push("shadowSampler" + lightIndex);
-    samplersList.push("depthSampler" + lightIndex);
+    samplersList.push(injectIndexInMiddle ? `shadow${lightIndex}Sampler` : "shadowSampler" + lightIndex);
+    samplersList.push(injectIndexInMiddle ? `depth${lightIndex}Sampler` : "depthSampler" + lightIndex);

```

Comment:
Given the comments above regarding the naming of the samplers ("sampler" in the WebGPU context, not "texture"!), these changes should not be needed (the sampler name should be `"shadowSampler" + lightIndex + WebGPUShaderProcessor.AutoSamplerSuffix` in the shader code).